subsys/fs/zms/zms.c

```diff
@@ -1683,9 +1736,8 @@ ssize_t zms_calc_free_space(struct zms_fs *fs)
 	/* Let's look now for special cases where some sectors have only ATEs with
 	 * small data size.
 	 */
-	const uint32_t second_to_last_offset = (2 * fs->ate_size);
 
-	for (uint32_t i = 0; i < fs->sector_count; i++) {
+	for (i = 0; i < fs->sector_count; i++) {
```

Comment:
Why not having kept those declarations that were in `for` loops? IMO it makes most sense to have them here to reduce the variables' scope if they're not meant to be used outside of the loop.